lib/ansible/modules/cloud/amazon/route53.py

```diff
@@ -593,13 +600,14 @@ def main():
             identifier_in = str(identifier_in)
 
         if rset.type == type_in and decoded_name.lower() == record_in.lower() and rset.identifier == identifier_in:
+            if need_to_sort_records:
+                # Sort records
+                rset.resource_records = sorted(rset.resource_records)
```

Comment:
I don't understand what this changes, as line 621 just sorts the resource records again whether or not `need_to_sort_records` is true or not.

Reply:
This changes the `rset` object itself; it's needed because the comparison is done in line 621 between `rset` and `wanted_rset`. All the other lines (i.e. putting stuff into `record`) is done only for `state == 'get'` if I understand it correctly, and not used anywhere else (in particular, not for comparison).

I kept the sorting in line 621 because that used to be done for all record types (but as I said, it is only used for `state == 'get'`); I could have only sorted again if it wasn't sorted here already, but I thought that would make it unnecessary complex.

Reply:
Right, I nearly wrote another comment about how I still don't understand but I got there while writing it :)